frontend/src/contexts/auth.context.tsx

```diff
@@ -63,12 +63,15 @@ export const AuthProvider = ({ children }: AuthProviderProps): JSX.Element => {
   };
   const authRedirection = async (isAuthenticated: boolean) => {
     if (isAuthenticated) {
+      const redirectUrl = router.query.redirect?.toString();
```

Comment:
This is a minor improvement, I suggest we check if `router.query.redirect` is an array, and if so, use only the first value. This may happen in case some code or a link generator mistakenly adds the same query param more than once. For example, `/login?redirect=/page1&redirect=/page2`
```suggestion
      let redirectUrl = router.query.redirect;

      if (Array.isArray(redirectUrl)) {
        redirectUrl = redirectUrl[0]?.toString();
      }
```

Reply:
Thank you for the feedback, updates are pushed :tada: